include/seastar/core/iostream.hh

```diff
@@ -151,6 +159,23 @@ public:
     }
 
 protected:
+#if SEASTAR_API_LEVEL >= 9
+    template <typename Fn>
+    requires std::is_invocable_r_v<future<>, Fn, temporary_buffer<char>&&>
+    static future<> fallback_put(std::span<temporary_buffer<char>> bufs, Fn fn) {
+        if (bufs.size() == 1) [[likely]] {
+            return fn(std::move(bufs.front()));
+        }
+
+        auto f = fn(std::move(bufs.front()));
+        for (auto&& buf : bufs.subspan(1)) {
+            f = std::move(f).then([fn, buf = std::move(buf)] () mutable {
```

Comment:
Doesn't this violate the contract? Nothing holds bufs safe.

I'd like the name of the function to better help the caller understand when to call it.

Reply:
Ah, the name already exists.

Reply:
Well, it should inherit the comment.

Reply:
> Doesn't this violate the contract? Nothing holds bufs safe.

It does not violate the contract, it creates a large future chain.

Reply:
Can be written as: std::ranges::fold_left(bufs | std::views::as_rvalue, make_ready_future<>(), []  (future<>, temporary_buffer) { ... });

Reply:
Yes, but only after we stop supporting C++20 :man_shrugging: